crates/noirc_frontend/src/hir/def_map/mod.rs

```diff
@@ -131,19 +131,19 @@ impl CrateDefMap {
         interner: &'a NodeInterner,
     ) -> impl Iterator<Item = TestFunction> + 'a {
         self.modules.iter().flat_map(|(_, module)| {
-            module
-                .value_definitions()
-                .filter_map(|id| id.as_function())
-                .filter(|id| {
-                    matches!(interner.function_meta(id).attributes, Some(Attribute::Test { .. }))
-                })
-                .map(|id| match interner.function_meta(&id).attributes {
-                    Some(Attribute::Test { expect_failure }) => {
-                        TestFunction::new(id, expect_failure)
+            module.value_definitions().filter_map(|id| {
+                if id.as_function().is_none() {
+                    return None;
+                } else {
+                    match interner.function_meta(&id.as_function().unwrap()).attributes {
```

Comment:
When doing the above suggestion, you won't need to call the `as_function()` again and no need to unwrap